packages/@yoda/httpdns/src/httpdns.cc

```diff
@@ -0,0 +1,158 @@
+#include "httpdns.h"
+#include <node_api.h>
+#include <uv.h>
+#include <stdio.h>
+#include <common.h>
+#include <string.h>
+#include <mutex>
+#include <list>
+
+using namespace std;
+static bool g_httpdns_initd = false;
+static uv_async_t async;
+
+struct HttpdnsAsyncTask {
+  napi_env env;
+  napi_ref callback;
+  uv_async_t async_handle;
+  int status;
+  ~HttpdnsAsyncTask() {
+    if (env && callback) {
+      NAPI_CALL_RETURN_VOID(env, napi_delete_reference(env, callback));
+    }
+  }
+};
+
+static int cb(int status, void* userdata) {
```

Comment:
`cb` looks too generic, could you please put this function on some other class or struct?